server/src/main/java/org/apache/druid/segment/join/BroadcastTableJoinableFactory.java

```diff
@@ -76,4 +76,27 @@ public boolean isDirectlyJoinable(DataSource dataSource)
     }
     return Optional.empty();
   }
+
+  @Override
+  public Optional<byte[]> computeJoinCacheKey(DataSource dataSource)
```

Comment:
seems like this and `build` could share a method that gets the `ReferenceCountingIndexedTable` if possible, and build can make the joinable and this method compute the key on the results

Reply:
Yes. good point.